### PR TITLE
ROX-18208: rhacs operator upgrade failed 4.1

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Cherry-pick of https://github.com/stackrox/stackrox/pull/6800

Customers who are using the kernel module collection method are unable to upgrade as that option has option has been removed in 4.1. This adds kernel module as a valid collection method. If someone tries to use kernel module as a collection method, the collection method will be changed to EBPF.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Manual operator upgrade works when using the downstream build based on these changes
- [x] Operator e2e test fails on master branch when the collection method is set to KernelModule
- [x] Operator e2e test passes when the collection method is set to KernelModule with these changes
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

### Manual Upgrade Tests

Followed the directions [here](https://gitlab.cee.redhat.com/stackrox/downstream-utils/-/blob/main/testing/books/operator-catalog.ipynb) and [here](https://docs.engineering.redhat.com/display/StackRox/Managing+a+Downstream+RHACS+Release+on+Red+Hat+Infrastructure#ManagingaDownstreamRHACSReleaseonRedHatInfrastructure-test-prerequisitesTestPrerequisites) with some changes.

Much more detailed steps and results are given [here](https://docs.google.com/document/d/1eEYYnsPX1JjCeDc5wIUE3p5B6QH8ZnBAt-YA-t7UvII/edit)

A quick summary of the results is that the upgrade from 4.0.2 to 4.1.0 failed when the collection method was KernelModule. The upgrade succeeded for 4.0.2 to 4.1.1 for the same steps. The collection method in the collector pods was switched to EBPF, while the UI still showed KernelModule. An attempt was also made to perform the following upgrade path 4.0.2 -> 4.1.0 -> 4.1.1. This did not succeed as it was not possible to upgrade either 4.0.2 or 4.1.0 after the failed upgrade to 4.1.0. This was tried with both a manual upgrade and an automatic upgrade.

### Branch with KernelModule in Test CR but without KernelModule in Collection Method Enum

In this case the operator e2e tests failed as expected.

https://github.com/stackrox/stackrox/pull/6863

### Branch with KernelModule in Test CR and with KernelModule in Collection Method Enum

The operator e2e tests passed in this case 

https://github.com/stackrox/stackrox/pull/6862

These two tests show that the changes made in this PR enable the upgrade to succeed when the collection method is set to KernelModule.

### CODE REVIEW

File: operator/config/crd/bases/platform.stackrox.io_securedclusters.yaml [hunk 1]
```diff
@@ -404,6 +404,7 @@ spec:
                         - EBPF
                         - CORE_BPF
                         - NoCollection
+                        - KernelModule
```
Comment (SimonBaeumer) [inline operator/config/crd/bases/platform.stackrox.io_securedclusters.yaml:407]: Is it automatically switched to eBPF in code when a customer selects `KernelModule`?
I am worried that customer uses `KernelModule` and the feature was removed from ACS.

**Review by SimonBaeumer — APPROVED** [review]

File: operator/config/crd/bases/platform.stackrox.io_securedclusters.yaml (hunk 1 above)
Comment (vladbologa) [reply operator/config/crd/bases/platform.stackrox.io_securedclusters.yaml:407]: Collector switches automatically to eBPF, I tested this.

**Pull request merged**